### PR TITLE
Add support for using remote source in MediaStep

### PR DESCRIPTION
**Pull request by lukaskabrt**
Right now `MediaStep` supports a file saved in the same file provider or a base64 encoded file as the source. This PR adds the third option - the file from a remote URL.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Media/Recipes/MediaStep.cs [hunk 1]
```diff
@@ -13,6 +14,7 @@ namespace OrchardCore.Media.Recipes
     public class MediaStep : IRecipeStepHandler
     {
         private readonly IMediaFileStore _mediaFileStore;
+        private readonly static HttpClient _httpClient = new HttpClient();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Media/Recipes/MediaStep.cs:17]: LGTM except if we can get the `HttpClient` from DI, but it's fine

**Review by deanmarcussen — APPROVED** [review]

**Comment by agriffard** [comment]
During the @sebastienros suggested to only use the SourcePath and check if it starts by `http`.

**Comment by lukaskabrt** [comment]
@agriffard if that's agreed on I can change it ... `SourceUrl` just seemed like a cleaner solution.

**Comment by agriffard** [comment]
Can you please create an issue and then a PR @lukaskabrt ?
@sebastienros also talked about the resiliency of the httpClient calling external resources.

**Pull request merged**